doc/tutorial/read_mapper/read_mapper_step2.cpp

```diff
@@ -0,0 +1,92 @@
+#include <seqan3/argument_parser/all.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/range/view/get.hpp>
+
+using namespace seqan3;
+//! [solution]
+//! [data_storage_t]
+struct data_storage_t
+{
+    concatenated_sequences<std::string> ref_id;
+    concatenated_sequences<std::vector<dna5>> ref_seq;
+    concatenated_sequences<std::string> query_id;
+    concatenated_sequences<std::vector<dna5>> query_seq;
+    concatenated_sequences<std::vector<phred42>> query_qual;
+};
+//! [data_storage_t]
+
+//! [read_input]
+void read_input(std::filesystem::path const & reference_path,
+                std::filesystem::path const & query_path,
+                data_storage_t & storage)
+//! [read_input]
+{
+    sequence_file_input reference_in{reference_path};
+    storage.ref_id  = std::move(get<field::ID>(reference_in));
+    storage.ref_seq = std::move(get<field::SEQ>(reference_in));
+    debug_stream << "Reference IDs: " << storage.ref_id << '\n';
+
+    sequence_file_input query_in{query_path};
+    storage.query_id   = std::move(get<field::ID>(query_in));
+    storage.query_seq  = std::move(get<field::SEQ>(query_in));
+    storage.query_qual = std::move(get<field::QUAL>(query_in));
+    debug_stream << "Number of queries: " << std::ranges::size(storage.query_id) << '\n';
```

Comment:
hm, we are always teaching them to use records-based I/O, now you use the columns-based I/O and even moving stuff around. I don't think it is very intuitive...

Reply:
I can change it. On the other hand, it's explained in the docs. And we also use `std::move` in the tutorial when we want to store them somewhere